File: 程式語言 V2答案檔/C++/308.cpp
#include <iostream>
using namespace std;

int compute(int);
int main()
{
    int i, n, fib[10];
    cin >> n;
    for (i = 0; i <= n; i++) {
        fib[i] = compute(i);
    }
    for (i = n; i >= 1; i--) {
        cout << "fib(" << i << ")=" << fib[i] << endl;
    }
    return (0);
}

int compute(int n)
{
    if (n == 0)
        return 0;
    else if (n == 1)
        return 1;
    else
        return compute(n - 1) + compute(n - 2);
}

File: 程式語言 V2答案檔/C#/308.cs
using System;

public class TQC308
{
    public static void Main(string[] args)
    {
        int n;
        n = int.Parse(Console.ReadLine());

        for (int i = n; i >= 1; i--) {
            Console.WriteLine(String.Format("fib({0})={1}", i, compute(i)));
        }
    }

    public static int compute(int n) {
        if (n == 0)
            return 0;
        else if (n == 1)
            return 1;
        
        return compute(n - 1) + compute(n - 2);
    }
}